eclair-core/src/main/scala/fr/acinq/eclair/wire/ChannelCodecs.scala

```diff
@@ -75,7 +75,7 @@ object ChannelCodecs extends Logging {
   val remoteParamsCodec: Codec[RemoteParams] = (
     ("nodeId" | publicKey) ::
       ("dustLimit" | satoshi) ::
-      ("maxHtlcValueInFlightMsat" | uint64) ::
+      ("maxHtlcValueInFlightMsat" | millisatoshi64) ::
```

Comment:
we should be able to use the `millisatoshi` codec here, since 21 million btc = 2100000000000000000 msat = 1d24b2dfac520000 msat, we don't need BigInt (but need to check potential impact on serialisation and storage)

Reply:
Okay then in the deserialization we would need truncate excess values (encoded in the uint64) to 2100000000000000000 msat

Reply:
Ah yes I see the pb now :( what happens if it is set to 0xFFFFFFFFFFFFFFFF ... I don't like truncating because with end up with `decode(encode(x)) != x`